### PR TITLE
Release 1.2.0

### PR DESCRIPTION
**Pull request by ggabernet**
- Added GitHub actions
- Fixed issues UMI barcode length and position
- Added downstream_only 
- For a full log of changes please check the `CHANGELOG.md`

## PR checklist

- [x] This comment contains a description of changes (with reason)
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If necessary, also make a PR on the [nf-core/bcellmagic branch on the nf-core/test-datasets repo]( https://github.com/nf-core/test-datasets/pull/new/nf-core/bcellmagic)
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [x] Make sure your code lints (`nf-core lint .`).
- [x] Documentation in `docs` is updated
- [x] `CHANGELOG.md` is updated
- [x] `README.md` is updated

**Learn more about contributing:** https://github.com/nf-core/bcellmagic/tree/master/.github/CONTRIBUTING.md

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,28 @@
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [1.2.0] - 2019-12-20 - "Riddikulus"
```
Comment (apeltzer) [inline CHANGELOG.md]:
```suggestion
## [1.2.0] - 2019-12-21 - "Riddikulus"
```

File: nextflow.config [hunk 2]
```diff
@@ -14,15 +14,18 @@ params {
   name = false
 
   // Initializing parameters
-  cprimers = false
-  vprimers = false
   set_cluster_threshold = false
-  cluster_threshold = false
+  cluster_threshold = -1
```
Comment (Leon-Bichmann) [inline nextflow.config]: Maybe use a more reasonable default value for this, in case sb. runs the pipeline without setting it?

File: main.nf [hunk 3]
```diff
@@ -305,25 +340,27 @@ process filter_by_sequence_quality {
 
     script:
     """
-    FilterSeq.py quality -s $r1 -q $filterseq_q --outname "${r1.baseName}" --log "${r1.baseName}_UMI_R1.log"
-    FilterSeq.py quality -s $r2 -q $filterseq_q --outname "${r2.baseName}" --log "${r2.baseName}_R2.log"
-    cp ".command.out" "${id}_command_log.txt"
+    FilterSeq.py quality -s $r1 -q 20 --outname "${r1.baseName}" --log "${r1.baseName}_UMI_R1.log" --nproc ${task.cpus} > "${id}_command_log.txt"
```
Comment (Leon-Bichmann) [inline main.nf]: Very small thing but I realize sometimes you use a single ">" and sometimes double ">>".

File: main.nf [hunk 4]
```diff
@@ -103,46 +110,61 @@ if( params.imgtdb_base ){
...
+if (params.umi_length == 0) {exit 1, "Please provide the UMI barcode length in the option `--umi_length`."}
+
+//Read processed tabs if downstream_only 
+if (params.downstream_only){
+    Channel
+        .fromFilePairs(params.changeo_tables, size: 1) {file -> file.baseName}
+        .ifEmpty {exit 1, "Cannot find any changeo tables matching: ${params.changeo_tables}.\nTry enclosing paths in quotes!\nTry adding a * wildcard!"}
+        .set {ch_tabs_for_clonal_analysis}
+        .println()
+    //ch_for_clonal_analysis = Channel.empty()
```
Comment (Leon-Bichmann) [inline main.nf]: rm comment?

File: main.nf [hunk 5]
```diff
@@ -665,21 +738,20 @@ process merge_tables{
...
     """
-
 }
 
 
 //Shazam! 
 process shazam{
     tag "${id}"    
-    publishDir "${params.outdir}/shazam/$id", mode: 'copy',
+    publishDir "${params.outdir}/genotyping/$id", mode: 'copy',
```
Comment (Leon-Bichmann) [inline main.nf]: ${id} ?

File: nextflow.config (hunk 2 above)
Comment (Leon-Bichmann) [reply nextflow.config]: 0.14?

File: docs/usage.md [hunk 6]
```diff
@@ -18,13 +18,19 @@
...
   * [`--igblast_base`](#--igblast_base)
   * [`--imgtdb_base`](#--imgtdb_base)
 * [Define clones](#Define-clones)
-  * [Manually set cluster threshold](#manually-set-cluster-threshold)
-  * [Only define clones](#only-define-clones)
+  * [`--set_cluster_threshold`](#--set_cluster_threshold)
+  * [`--cluster_threshold`](#--cluster_threshold)
+* [Downstream analysis](#Downstream-analysis)
+  * [`--downstream_only`](#--downstream_only)
+  * [`--skipDownstream`](#--skipDownstream)
```
Comment (Leon-Bichmann) [inline docs/usage.md]: changeo_tables and skipDownstram missing?

File: main.nf (hunk 5 above)
Comment (ggabernet) [reply main.nf]: $id also works here

File: main.nf (hunk 3 above)
Comment (ggabernet) [reply main.nf]: Depending if I want to append the text at the end of the file ;)

**Pull request merged**